backend/src/routes/api/notebooks/notebookUtils.ts

```diff
@@ -161,28 +162,29 @@ export const createNotebook = async (
     throw error;
   });
 
-  let route: Route;
+  let route: Route | void;
   let fetchTime = 10;
   while (fetchTime > 0) {
-    try {
-      route = await getRoute(fastify, namespace, notebookName);
-      return await patchNotebookRoute(fastify, route, namespace, notebookName);
-    } catch (e) {
+    route = await getRoute(fastify, namespace, notebookName).catch(async (e) => {
       // if the route is not created yet
       // wait for 1 sec and fetch time minus 1
-      if (e.response?.statusCode === 404) {
+      if (e.code === 404) {
         await new Promise((r) => setTimeout(r, 1000));
         fetchTime -= 1;
         if (fetchTime <= 0) {
           fastify.log.warn(
             'Wait for 10 seconds and route is still missing, skipping patching for now.',
           );
         }
-        continue;
+        return;
       }
       throw e;
-    }
+    });
   }
+  if (route) {
+    return await patchNotebookRoute(fastify, route, namespace, notebookName);
+  }
```

Comment:
Interesting... I guess why try to patch if we don't have a route 🤔 Maybe add a log after this to say we failed to patch it. Make it an error log -- let us just assume if it fails here, it should be a problem to catch the eye of the log reader. Our code having a fallback is separate to this function failing.